### PR TITLE
refactor($interval): do not use `notify` to trigger the callback

### LINKED ISSUE #12533: Drop non-compliant Promises/A+  progress/notify on $q
Makes it really hard to replace $q when using a compliant library, since angular uses progress/notify handlers on core itself.

### PR DESCRIPTION
**Pull request by lgalfaso**
Do not use `$q.notify` to trigger the callback. This allows `$q` to be replaced
with another Promise/A+ compilant library

Note: this PR is not enough to close #12533, but should solve the main issue in trying to use Bluebird as a replacement of `$q`

### CODE REVIEW

**Comment by Narretz** [comment]
If we are making this change to support a specific use case, shouldn't we add a test for that behavior? Otherwise someone else might refactor this back to what it was, and it would break again,

**Comment by lgalfaso** [comment]
@Narretz done!

File: src/ng/interval.js [hunk 1]
```diff
@@ -144,11 +144,14 @@ function $IntervalProvider() {
 
       count = isDefined(count) ? count : 0;
 
-      promise.then(null, null, (!hasParams) ? fn : function() {
-        fn.apply(null, args);
-      });
-
       promise.$$intervalId = setInterval(function tick() {
+        if (intervals[promise.$$intervalId]) {
```
Comment (gkalpak) [inline src/ng/interval.js]: Why is this check needed ?

File: src/ng/interval.js [hunk 2]
```diff
@@ -164,6 +167,14 @@ function $IntervalProvider() {
       intervals[promise.$$intervalId] = deferred;
 
       return promise;
+
+       function callback() {
```
Comment (gkalpak) [inline src/ng/interval.js]: Too indented...

File: src/ng/interval.js (hunk 1 above)
Comment (lgalfaso) [inline src/ng/interval.js]: you are right

File: src/ng/interval.js (hunk 2 above)
Comment (lgalfaso) [inline src/ng/interval.js]: thanks!

**Pull request closed without merging**